### PR TITLE
Revised and improved examples.

### PR DESCRIPTION
**Pull request by ali-ramadhan**
This PR improves the existing examples and updates them to work with the current version of the model.

There were also a couple of minor changes:
* Output writer API: Filename suffix is now called a file extension. No more inserting underscores for empty names and filename_prefix. User should include underscore if they want one.
* Model metadata:  Use `:CPU` and `:GPU` instead of `:cpu` and `:gpu`. This is so that Oceananigans.jl agrees with the symbols used in GPUifyLoops.jl.

### CODE REVIEW

**Comment by vchuravy** [comment]
> Model metadata: Use :CPU and :GPU instead of :cpu and :gpu. This is so that Oceananigans.jl agrees with the symbols used in GPUifyLoops.jl.

We probably should make these types

**Comment by glwagner** [comment]
> We probably should make these types

I like that idea! As simple as

```
abstract type Device end
struct CPU <: Device end
struct GPU <: Device end
```

in `GPUifyloops.jl`?

File: examples/barotropic_tracer_advection.jl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+include("utils.jl")
+
+Nx, Ny, Nz = 128, 128, 1
+Lx, Ly, Lz = 2000, 2000, 10
+Nt, Δt = 25, 1
+
+model = Model(N=(Nx, Ny, Nz), L=(Lx, Ly, Lz), ν=4e-2, κ=4e-2)
+
+@. model.tracers.T.data[70:90, 10:30, 1] += 0.01
```
Comment (glwagner) [inline examples/barotropic_tracer_advection.jl]: Should we get into the habit of setting fields with functions? It seems potentially more robust. for example, check out the [`set!` function in `OceanTurb.jl`](https://github.com/glwagner/OceanTurb.jl/blob/35ce00da3ef1bdd707a618bce6c7e61fe994d6c5/src/fields.jl#L132), which combined with [an overloading of `setproperty!`](https://github.com/glwagner/OceanTurb.jl/blob/35ce00da3ef1bdd707a618bce6c7e61fe994d6c5/src/OceanTurb.jl#L127) enables syntax like

```
T0(x, y, z) = 285 + 0.01*exp( -(x^2+y^2+z^2) )
model.solution.T = T0
```
For example.

Comment (ali-ramadhan) [reply examples/barotropic_tracer_advection.jl]: Yes that would be awesome, then we can set using functions, constants, arrays, profiles, etc.

**Pull request merged**